### PR TITLE
feat(COS-2441): Add docs methods onFielFocus, onFielBlur, onFieldSubm…

### PR DESCRIPTION
**Pull request by brahianzapata**

### CODE REVIEW

**Review by kushki-dflasso — CHANGES_REQUESTED** [review]

File: src/repository/ICard.ts [hunk 1]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
+   * onFieldFocus a field specify or form field
    *
-   * @function
+   * This event is emitted when the field gains focus, otherwise it will throw an exception
+   *
+   * @group Methods
    * @param {(FormValidity | FieldValidity) => void} event - The function called when the form field is focused
-   * @param {FieldTypeEnum} [fieldType] - The type of form field (optional)
-   * @returns {void}
+   * @param {FieldTypeEnum} fieldType - The type of field (optional)
```
[the remark this answers is not in the document]
Comment (brahianzapata) [reply src/repository/ICard.ts:134]: Siii es correcto

**Review by kushki-dflasso — APPROVED** [review]

File: src/repository/ICard.ts [hunk 2]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
   ): void;
 
   /**
-   * This event is emitted when the field gains focus
+   * Method onFieldFocus is listening all fields or one specify if have focus
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:128]: This event is emitted when the field gains focus

File: src/repository/ICard.ts [hunk 3]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
    *
-   * @function
-   * @param {(FormValidity | FieldValidity) => void} event - The function called when the form field is focused
-   * @param {FieldTypeEnum} [fieldType] - The type of form field (optional)
-   * @returns {void}
+   * This event is emitted when the field agains focus
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:130]: borrar

File: src/repository/ICard.ts [hunk 4]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
+   *          console.log("Form invalid", event);
+   *        }
+   *      });
+   *    // On Success, can get onFieldFocus, ex. FormValidity: { isFormValid: true, triggeredBy: cardholderName, fields: Fields}
+   *  } catch (error: any) {
+   *      console.error("Catch error on onFieldFocus", error.code, error.message);
+   *  }
+   * ```
+   *
+   * Handling event 'focus' of card holderName hosted field
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:157]: card holderName => an especific

File: src/repository/ICard.ts [hunk 5]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
+   * ```
+   *
    */
   onFieldFocus(
     event: (fieldEvent: FormValidity | FieldValidity) => void,
     fieldType?: FieldTypeEnum
   ): void;
 
   /**
+   * Method onFieldBlur is listening all fields or one specify if have blur
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:182]: borrar y solo dejar This event is emitted when the field loses focus

File: src/repository/ICard.ts [hunk 6]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
+   *          console.log("Form invalid", event);
+   *        }
+   *      });
+   *    // On Success, can get onFieldBlur, ex. FormValidity: { isFormValid: true, triggeredBy: cardholderName, fields: Fields}
+   *  } catch (error: any) {
+   *      console.error("Catch error on onFieldBlur", error.code, error.message);
+   *  }
+   * ```
+   *
+   * Handling event 'blur' of card holderName hosted field
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:211]: card holderName => an especific

File: src/repository/ICard.ts [hunk 7]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
     fieldType?: FieldTypeEnum
   ): void;
 
   /**
-   * This event is emitted when the field has submit
-   * @function
-   * @param {(FormValidity | FieldValidity) => void} event - The function called when the form field is submitted
-   * @param {FieldTypeEnum} [fieldType] - The type of form field (optional)
-   * @returns {void}
+   * Method onFieldSubmit is listening all fields or one specify if have submit
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:236]: borrar

File: src/repository/ICard.ts [hunk 8]
```diff
@@ -125,154 +125,220 @@ export interface ICard {
...
+   *          console.log("Form invalid", event);
+   *        }
+   *      });
+   *    // On Success, can get onFieldSubmit, ex. FormValidity: { isFormValid: true, triggeredBy: cardholderName, fields: Fields}
+   *  } catch (error: any) {
+   *      console.error("Catch error on onFieldSubmit", error.code, error.message);
+   *  }
+   * ```
+   *
+   * Handling event 'submit' of card holderName hosted field
```
Comment (esteban-aristizabal) [inline src/repository/ICard.ts:265]: card holderName => an especific

File: README.md [hunk 9]
```diff
@@ -43,32 +43,33 @@ Use a script tag inside your page to add the feature. When adding the following
...
 
 ```ts
-// Example usage of the init function
-const options = {
-  apiKey: 'YOUR_API_KEY',
-  environment: 'sandbox'
+import { IKushki, init, KushkiError } from "Kushki";
+
+const kushkiOptions : KushkiOptions = {
+  publicCredentialId: '<public-credential-id>', // This corresponds to the public merchant credential
```
Comment (esteban-aristizabal) [inline README.md:54]: This corresponds to the public merchant credential => This corresponds to the public credential of the merchant

**Review by esteban-aristizabal — CHANGES_REQUESTED** [review]

**Review by esteban-aristizabal — APPROVED** [review]

**Pull request merged**